src/wasm-lib/kcl/src/coredump/mod.rs

```diff
@@ -95,45 +132,45 @@ pub struct AppInfo {
     pub timestamp: chrono::DateTime<chrono::Utc>,
     /// If the app is running in tauri or the browser.
     pub tauri: bool,
-
     /// The os info.
     pub os: OsInfo,
-
     /// The webrtc stats.
     pub webrtc_stats: WebrtcStats,
-
     /// A GitHub issue url to report the core dump.
-    /// This gets prepoulated with all the core dump info.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub github_issue_url: Option<String>,
-
     /// Engine pool the client is connected to.
     pub pool: String,
+    /// The client state (singletons and xstate).
+    pub client_state: JValue,
 }
 
-impl AppInfo {
+impl CoreDumpInfo {
     /// Set the github issue url.
-    pub fn set_github_issue_url(&mut self, screenshot_url: &str) -> Result<()> {
+    pub fn set_github_issue_url(&mut self, screenshot_url: &str, coredump_url: &str, coredump_id: &Uuid) -> Result<()> {
+        let coredump_filename = Path::new(coredump_url).file_name().unwrap().to_str().unwrap();
         let tauri_or_browser_label = if self.tauri { "tauri" } else { "browser" };
         let labels = ["coredump", "bug", tauri_or_browser_label];
         let body = format!(
-            r#"[Insert a description of the issue here]
+            r#"[Add a title above and insert a description of the issue here]
 
 ![Screenshot]({})
 
 <details>
 <summary><b>Core Dump</b></summary>
 
-```json
-{}
-```
+[{}]({})
+
+Reference ID: {}
 </details>
 "#,
-            screenshot_url,
-            serde_json::to_string_pretty(&self)?
+            screenshot_url, coredump_filename, coredump_url, coredump_id
```

Comment:
I think this would be more readable if the variables here were interpolated directly e.g. `Reference ID: {coredump_id}`

Reply:
Updated in 1a2f6ec2